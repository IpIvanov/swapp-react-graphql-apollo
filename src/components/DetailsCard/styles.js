const styles = () => ({
  card: {
    display: 'flex',
    flexDirection: 'row',
    marginBottom: 20,
  },
  content: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
  },
  description: {
    textAlign: 'left',
    marginBottom: 10,
    padding: 0,
  },
  cover: {
    width: '100%',
    height: 420,
  },
  starshipCover: {
    width: 120,
  },
  title: {
    color: '#4BD5EE',
    fontWeight: 700,
  },
});

export default styles;
