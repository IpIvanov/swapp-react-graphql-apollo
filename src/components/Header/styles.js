const styles = (theme) => ({
  container: (props) => ({
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 5,
    border: `1px soldi ${theme.palette[props.theme].AppBar.borderColor}`,
    background: theme.palette[props.theme].AppBar.backgroundColor,
  }),
  rightHeader: {
    display: 'flex',
    alignItems: 'center',
  },
  link: (props) => ({
    padding: 10,
    textDecoration: 'none',
    color: theme.palette[props.theme].AppBar.fontColor,
    '&:hover': {
      cursor: 'pointer',
    },
  }),
  linkHome: {
    padding: 10,
    textDecoration: 'none',
    fontFamily: 'SfDistantGalaxyOutline',
    color: theme.palette.commonColors.yellow,
    fontSize: 26,
    '&:hover': {
      cursor: 'pointer',
    },
  },
  exit: (props) => ({
    padding: 3,
    borderRadius: 3,
    textDecoration: 'none',
    backgroundColor: '#fff',
    color: theme.palette[props.theme].AppBar.fontColor,
    '&:hover': {
      cursor: 'pointer',
    },
  }),
  active: {
    textDecoration: 'none',
  },
});

export default styles;
