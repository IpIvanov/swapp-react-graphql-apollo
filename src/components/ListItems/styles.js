const styles = (theme) => ({
  card: {
    display: 'flex',
    flexDirection: 'row',
    marginBottom: 20,
  },
  content: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  cover: {
    width: 100,
    height: 120,
  },
  name: (props) => ({
    color: theme.palette[props.theme].PrimaryHeading.fontColor,
    fontFamily: 'SfDistantGalaxyOutline',
    textTransform: 'none',
    fontSize: 22,
  }),
  button: (props) => ({
    textTransform: 'none',
    backgroundColor: theme.palette[props.theme].OutlineButtons.backgroundColor,
    borderColor: theme.palette[props.theme].OutlineButtons.borderColor,
    color: theme.palette[props.theme].OutlineButtons.fontColor,
    fontSize: 20,
    fontWeight: 900,
  }),
  linkTo: {
    textDecoration: 'none',
  },
});

export default styles;
