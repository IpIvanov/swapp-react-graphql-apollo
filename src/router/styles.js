const styles = (theme) => ({
  container: (props) => ({
    color: theme.palette[props.theme].textColor,
    backgroundColor: theme.palette[props.theme].backgroundColor,
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    flexDirection: 'column',
    alignContent: 'center',
    minHeight: '100vh',
    minWidth: '100%',
    textAlign: 'center',
  }),
});

export default styles;
