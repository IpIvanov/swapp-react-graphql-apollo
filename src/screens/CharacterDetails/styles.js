const styles = (theme) => ({
  wrapper: {
    minWidth: '100%',
  },
  container: {
    marginTop: 50,
  },
  name: (props) => ({
    color: theme.palette[props.theme].PrimaryHeading.fontColor,
    fontFamily: 'SfDistantGalaxyOutline',
    fontSize: 36,
  }),
  starshipsTitle: {
    fontFamily: 'SfDistantGalaxyOutline',
    fontSize: 36,
  },
  divider: { marginBottom: 20 },
});

export default styles;
