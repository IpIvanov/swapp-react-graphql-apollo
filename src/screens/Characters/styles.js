const styles = () => ({
  wrapper: {
    minWidth: '100%',
  },
  container: {
    marginTop: 50,
  },
});

export default styles;
