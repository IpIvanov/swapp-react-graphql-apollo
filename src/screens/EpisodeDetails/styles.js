const styles = (theme) => ({
  wrapper: {
    minWidth: '100%',
  },
  container: {
    marginTop: 50,
  },
  card: {
    display: 'flex',
    flexDirection: 'row',
    marginBottom: 20,
  },
  content: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
  },
  description: {
    textAlign: 'left',
    marginBottom: 10,
  },
  movieCover: {
    width: 300,
    height: 300,
  },
  name: (props) => ({
    color: theme.palette[props.theme].PrimaryHeading.fontColor,
    fontFamily: 'SfDistantGalaxyOutline',
    textTransform: 'none',
    fontSize: 22,
  }),
  textColor: (props) => ({ color: theme.palette[props.theme].PrimaryHeading.fontColor }),
});

export default styles;
