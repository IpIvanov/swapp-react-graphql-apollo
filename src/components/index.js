import Header from './Header';
import PrivateRoute from './PrivateRoute';
import ListItems from './ListItems';
import DetailsCard from './DetailsCard';

export { Header, PrivateRoute, ListItems, DetailsCard };
