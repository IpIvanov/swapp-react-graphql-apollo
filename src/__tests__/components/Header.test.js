/* eslint global-require: 0 */
import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from '@material-ui/core/styles';
import ThemeContext from '../../contexts/ThemeContext';
import { Header } from '../../components';

// const mountWithTheme = (children) => mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
// const renderWithTheme = (children) => render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
// const shallowWithTheme = (children) => shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

jest.mock('@apollo/react-hooks', () => ({
  useApolloClient: () => () => ({
    writeData: jest.fn(),
  }),
}));

describe('Components | Header', () => {
  let theme;

  beforeEach(() => {
    jest.spyOn({ preventDefault: () => { } }, 'preventDefault');
    const localStorageMock = {
      getItem: jest.fn(),
      setItem: jest.fn(),
      clear: jest.fn(),
    };
    global.localStorage = localStorageMock;

    theme = require('../../theme');
  });

  it('renders without crashing', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme.default}>
        <ThemeContext.Provider value={{ theme: 'dark', setTheme: () => { } }}>
          <Header />
        </ThemeContext.Provider>
      </ThemeProvider>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
