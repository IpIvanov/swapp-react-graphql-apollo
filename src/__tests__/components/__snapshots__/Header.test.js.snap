// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | Header renders without crashing 1`] = `
<div
  className="makeStyles-container-1 makeStyles-container-7"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-linkHome-4"
    onClick={[Function]}
  >
    <ForwardRef(Typography)
      className="makeStyles-linkHome-4"
      classes={
        Object {
          "alignCenter": "MuiTypography-alignCenter",
          "alignJustify": "MuiTypography-alignJustify",
          "alignLeft": "MuiTypography-alignLeft",
          "alignRight": "MuiTypography-alignRight",
          "body1": "MuiTypography-body1",
          "body2": "MuiTypography-body2",
          "button": "MuiTypography-button",
          "caption": "MuiTypography-caption",
          "colorError": "MuiTypography-colorError",
          "colorInherit": "MuiTypography-colorInherit",
          "colorPrimary": "MuiTypography-colorPrimary",
          "colorSecondary": "MuiTypography-colorSecondary",
          "colorTextPrimary": "MuiTypography-colorTextPrimary",
          "colorTextSecondary": "MuiTypography-colorTextSecondary",
          "displayBlock": "MuiTypography-displayBlock",
          "displayInline": "MuiTypography-displayInline",
          "gutterBottom": "MuiTypography-gutterBottom",
          "h1": "MuiTypography-h1",
          "h2": "MuiTypography-h2",
          "h3": "MuiTypography-h3",
          "h4": "MuiTypography-h4",
          "h5": "MuiTypography-h5",
          "h6": "MuiTypography-h6",
          "noWrap": "MuiTypography-noWrap",
          "overline": "MuiTypography-overline",
          "paragraph": "MuiTypography-paragraph",
          "root": "MuiTypography-root",
          "srOnly": "MuiTypography-srOnly",
          "subtitle1": "MuiTypography-subtitle1",
          "subtitle2": "MuiTypography-subtitle2",
        }
      }
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root makeStyles-linkHome-4 MuiTypography-body1"
        onClick={[Function]}
      >
        SWAPP
      </p>
    </ForwardRef(Typography)>
  </WithStyles(ForwardRef(Typography))>
  <div
    className="makeStyles-rightHeader-2"
  >
    <WithStyles(ForwardRef(Link))
      className="makeStyles-link-3 makeStyles-link-8"
      href="/"
      onClick={[Function]}
      underline="none"
    >
      <ForwardRef(Link)
        className="makeStyles-link-3 makeStyles-link-8"
        classes={
          Object {
            "button": "MuiLink-button",
            "focusVisible": "Mui-focusVisible",
            "root": "MuiLink-root",
            "underlineAlways": "MuiLink-underlineAlways",
            "underlineHover": "MuiLink-underlineHover",
            "underlineNone": "MuiLink-underlineNone",
          }
        }
        href="/"
        onClick={[Function]}
        underline="none"
      >
        <WithStyles(ForwardRef(Typography))
          className="MuiLink-root MuiLink-underlineNone makeStyles-link-3 makeStyles-link-8"
          color="primary"
          component="a"
          href="/"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          variant="inherit"
        >
          <ForwardRef(Typography)
            className="MuiLink-root MuiLink-underlineNone makeStyles-link-3 makeStyles-link-8"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter",
                "alignJustify": "MuiTypography-alignJustify",
                "alignLeft": "MuiTypography-alignLeft",
                "alignRight": "MuiTypography-alignRight",
                "body1": "MuiTypography-body1",
                "body2": "MuiTypography-body2",
                "button": "MuiTypography-button",
                "caption": "MuiTypography-caption",
                "colorError": "MuiTypography-colorError",
                "colorInherit": "MuiTypography-colorInherit",
                "colorPrimary": "MuiTypography-colorPrimary",
                "colorSecondary": "MuiTypography-colorSecondary",
                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                "displayBlock": "MuiTypography-displayBlock",
                "displayInline": "MuiTypography-displayInline",
                "gutterBottom": "MuiTypography-gutterBottom",
                "h1": "MuiTypography-h1",
                "h2": "MuiTypography-h2",
                "h3": "MuiTypography-h3",
                "h4": "MuiTypography-h4",
                "h5": "MuiTypography-h5",
                "h6": "MuiTypography-h6",
                "noWrap": "MuiTypography-noWrap",
                "overline": "MuiTypography-overline",
                "paragraph": "MuiTypography-paragraph",
                "root": "MuiTypography-root",
                "srOnly": "MuiTypography-srOnly",
                "subtitle1": "MuiTypography-subtitle1",
                "subtitle2": "MuiTypography-subtitle2",
              }
            }
            color="primary"
            component="a"
            href="/"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            variant="inherit"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineNone makeStyles-link-3 makeStyles-link-8 MuiTypography-colorPrimary"
              href="/"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Episodes
            </a>
          </ForwardRef(Typography)>
        </WithStyles(ForwardRef(Typography))>
      </ForwardRef(Link)>
    </WithStyles(ForwardRef(Link))>
    <WithStyles(ForwardRef(Link))
      className="makeStyles-link-3 makeStyles-link-8"
      href="/"
      onClick={[Function]}
      underline="none"
    >
      <ForwardRef(Link)
        className="makeStyles-link-3 makeStyles-link-8"
        classes={
          Object {
            "button": "MuiLink-button",
            "focusVisible": "Mui-focusVisible",
            "root": "MuiLink-root",
            "underlineAlways": "MuiLink-underlineAlways",
            "underlineHover": "MuiLink-underlineHover",
            "underlineNone": "MuiLink-underlineNone",
          }
        }
        href="/"
        onClick={[Function]}
        underline="none"
      >
        <WithStyles(ForwardRef(Typography))
          className="MuiLink-root MuiLink-underlineNone makeStyles-link-3 makeStyles-link-8"
          color="primary"
          component="a"
          href="/"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          variant="inherit"
        >
          <ForwardRef(Typography)
            className="MuiLink-root MuiLink-underlineNone makeStyles-link-3 makeStyles-link-8"
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter",
                "alignJustify": "MuiTypography-alignJustify",
                "alignLeft": "MuiTypography-alignLeft",
                "alignRight": "MuiTypography-alignRight",
                "body1": "MuiTypography-body1",
                "body2": "MuiTypography-body2",
                "button": "MuiTypography-button",
                "caption": "MuiTypography-caption",
                "colorError": "MuiTypography-colorError",
                "colorInherit": "MuiTypography-colorInherit",
                "colorPrimary": "MuiTypography-colorPrimary",
                "colorSecondary": "MuiTypography-colorSecondary",
                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                "displayBlock": "MuiTypography-displayBlock",
                "displayInline": "MuiTypography-displayInline",
                "gutterBottom": "MuiTypography-gutterBottom",
                "h1": "MuiTypography-h1",
                "h2": "MuiTypography-h2",
                "h3": "MuiTypography-h3",
                "h4": "MuiTypography-h4",
                "h5": "MuiTypography-h5",
                "h6": "MuiTypography-h6",
                "noWrap": "MuiTypography-noWrap",
                "overline": "MuiTypography-overline",
                "paragraph": "MuiTypography-paragraph",
                "root": "MuiTypography-root",
                "srOnly": "MuiTypography-srOnly",
                "subtitle1": "MuiTypography-subtitle1",
                "subtitle2": "MuiTypography-subtitle2",
              }
            }
            color="primary"
            component="a"
            href="/"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            variant="inherit"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineNone makeStyles-link-3 makeStyles-link-8 MuiTypography-colorPrimary"
              href="/"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Characters
            </a>
          </ForwardRef(Typography)>
        </WithStyles(ForwardRef(Typography))>
      </ForwardRef(Link)>
    </WithStyles(ForwardRef(Link))>
    <div
      className="makeStyles-exit-5 makeStyles-exit-9"
    >
      <ForwardRef
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(SvgIcon))
          onClick={[Function]}
        >
          <ForwardRef(SvgIcon)
            classes={
              Object {
                "colorAction": "MuiSvgIcon-colorAction",
                "colorDisabled": "MuiSvgIcon-colorDisabled",
                "colorError": "MuiSvgIcon-colorError",
                "colorPrimary": "MuiSvgIcon-colorPrimary",
                "colorSecondary": "MuiSvgIcon-colorSecondary",
                "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                "root": "MuiSvgIcon-root",
              }
            }
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              onClick={[Function]}
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M21 3.01H3c-1.1 0-2 .9-2 2V9h2V4.99h18v14.03H3V15H1v4.01c0 1.1.9 1.98 2 1.98h18c1.1 0 2-.88 2-1.98v-14c0-1.11-.9-2-2-2zM11 16l4-4-4-4v3H1v2h10v3z"
              />
            </svg>
          </ForwardRef(SvgIcon)>
        </WithStyles(ForwardRef(SvgIcon))>
      </ForwardRef>
    </div>
  </div>
</div>
`;
